tools/wt_binary_decode.py

```diff
@@ -722,22 +829,65 @@ def row_decode(p, b, pagehead, blockhead):
                 pfx_compress_byte = b.read(1)[0]
                 s = 'short key prefix={} {} bytes'.format(hex(pfx_compress_byte), l)
                 x = b.read(l)
+                pagestats.num_keys += 1
+                pagestats.keys_sz += l
             else: # short is 1 or 3
                 l = (desc & 0xfc) >> 2
                 if short == 1:
                     s = 'short key {} bytes'.format(l)
+                    pagestats.num_keys += 1
+                    pagestats.keys_sz += l
                 else:
                     s = 'short val {} bytes'.format(l)
                 x = b.read(l)
 
-            if s != '?':
-                p.rint(f'{desc_str}{s}:')
-                p.rint(raw_bytes(x))
-            else:
-                dumpraw(p, b, cellpos)
+            #if s != '?':
```

Comment:
All of the places I commented out the prints, we should instead update it to use opts.no_data. It can't be on the printer because I think its good that errors go out. But for print_timestamps and row_decode we should be more restricted on what we print with that flag.

Reply:
opts.no_data skips `row_decode` entirely I think, which is a bit annoying but for now I don't think we need to actually hide this

Reply:
I guess we could add a `--verbose` or something

Reply:
I think that could work. We don't actually want to print all this out while doing the csv output, and we don't want to lose it from the script.

Reply:
I added `--skip-data` to keep the performance benefit when that's wanted and made `--no-data` -> `--hide-data` and fixed the printer logic.

Reply:
We should debug that some more (as to why the printing is causing the parsing to fail). I've got some time to take a look.

Reply:
It is pre-existing. The difference is the -D flag includes the exceptions so it looks worse than it actually is. I think we should leave these uncommented but catch the common exception and comment out printing them.

Reply:
Discussed offline with Jonas. We should uncomment these, but add a try/except around the p.rint(raw_bytes(x)) which generates the majority of the failures. The printing of the exception (or raising it higher as another choice) can be commented out with a separate ticket to diagnose that further. That shouldn't cause any follow on issues has its handling bytes that have already been read from the buffer.